firebase-database/src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java

```diff
@@ -559,12 +559,17 @@ public void onDisconnect(Connection.DisconnectReason reason) {
 
   @Override
   public void onKill(String reason) {
-    logger.warn(
-        "Firebase Database connection was forcefully killed by the server. Will not attempt"
-            + " reconnect. Reason: "
-            + reason);
+    if (reason.equals("Invalid AppCheck token")) {
+      logger.warn("Detected invalid AppCheck token.Reconnecting");
```

Comment:
I wonder if we should increment and check `invalidAppCheckTokenCount` here as well to avoid an infinite reconnection loop in case e.g. when an App Check provider is not configured?